### DISCUSSION TITLE
Buttons do not seem to get a unique on_click?

### DISCUSSION THREAD

**Discussion by thompsm** [zauberzeug/nicegui#614] [Q&A] [answered]
Hi,

I'm trying to create some buttons that can redirect me to different pages, but the on_click callback's all seem to be the same? Or at least pass the same parameter? Is this possible to do please?

Here's some simple test code:

```py
from nicegui import ui

pages = ["1", "2", "3", "4"]

@ui.page('/page_details/{page_id}')
def page_details(page_id):
    ui.label('Page id is ' +  page_id)
    ui.button('Back', on_click=lambda: ui.open("/"))

with ui.row():
    for pageId in pages:
        ui.button(pageId, on_click=lambda: ui.open("/page_details/" + pageId))

ui.run()
```

When I run this, all the buttons send me to number 4? Or whatever the last number happens to be in the pages list.

**Comment by falkoschindler** [comment, marked as answer]
Hi @thompsm! This is a common pitfall when working with Python's lambda statements.
See here for a very similar case: https://github.com/zauberzeug/nicegui/discussions/471
I'll close this discussion because it's basically a duplicate. 🙂

**Reply by thompsm** [reply]
Thanks! That's got it working.

**Comment by thompsm** [comment]
Updated test code that now works for anyone else new to lambdas like me.

```py
from nicegui import ui

pages = ["1", "2", "3", "4"]

@ui.page('/page_details/{page_id}')
def page_details(page_id):
    ui.label('Page id is ' +  page_id)
    ui.button('Back', on_click=lambda: ui.open("/"))

with ui.row():
    for pageId in pages:
        ui.button(pageId, on_click=lambda e, pageId=pageId: ui.open("/page_details/" + pageId))

ui.run()
```
